python/DEVELOPMENT.md

```diff
@@ -0,0 +1,121 @@
+<!---
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License. See accompanying LICENSE file.
+-->
+
+## Developer guide for conda users
+
+First, set up your thirdparty C++ toolchain using libraries from conda-forge:
+
+```shell
+conda config --add channels conda-forge
+
+export ARROW_BUILD_TYPE=Release
+
+export CPP_TOOLCHAIN=$HOME/cpp-toolchain
+export LD_LIBRARY_PATH=$CPP_TOOLCHAIN/lib:$LD_LIBRARY_PATH
+
+export BOOST_ROOT=$CPP_TOOLCHAIN
+export FLATBUFFERS_HOME=$CPP_TOOLCHAIN
+export RAPIDJSON_HOME=$CPP_TOOLCHAIN
+export THRIFT_HOME=$CPP_TOOLCHAIN
+export ZLIB_HOME=$CPP_TOOLCHAIN
+export SNAPPY_HOME=$CPP_TOOLCHAIN
+export BROTLI_HOME=$CPP_TOOLCHAIN
+export JEMALLOC_HOME=$CPP_TOOLCHAIN
+export ARROW_HOME=$CPP_TOOLCHAIN
+export PARQUET_HOME=$CPP_TOOLCHAIN
+
+conda create -y -q -p $CPP_TOOLCHAIN \
+      flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib brotli jemalloc
+```
+
+Now, activate a conda environment containing your target Python version and
+NumPy installed:
+
+```shell
+conda create -y -q -n pyarrow-dev python=3.6 numpy
+source activate pyarrow-dev
+```
+
+Now build and install the Arrow C++ libraries from the base of your
+apache/arrow clone:
+
+```shell
+mkdir cpp/build
+pushd cpp/build
+
+cmake -DCMAKE_BUILD_TYPE=$ARROW_BUILD_TYPE \
+      -DCMAKE_INSTALL_PREFIX=$CPP_TOOLCHAIN \
+      -DARROW_PYTHON=on \
+      -DARROW_BUILD_TESTS=OFF \
+      ..
+make -j4
```

Comment:
OK, it looks like you have an artifact of a prior Arrow install in your /usr/local, and `pkg_check_modules` is picking it up here https://github.com/apache/parquet-cpp/blob/master/cmake_modules/ThirdpartyToolchain.cmake#L360. Can you delete

```
rm -rf /usr/local/lib/libarrow*
rm -rf /usr/local/lib/pkgconfig/arrow*
```

and try again? I'm opening a Parquet JIRA so that `$ARROW_HOME` is respected if it is set in the environment.